README.md

```diff
@@ -1,8 +1,248 @@
+
+- **ALPHA** (Elite): Best individuals with low mutation rates for exploitation
+- **BETA** (High performers): Good solutions with moderate diversity
+- **GAMMA** (Average): Balanced exploration and exploitation
+- **DELTA** (Below average): Higher mutation for local search
+- **EPSILON** (Diverse): Highest mutation rates for exploration and diversity
+
+## 📋 Basic Usage
+
+### Minimal Example
+
+```julia
+using Pkg
+Pkg.activate(".")  # If using the downloaded repository
+
+include("src/BraveNewAlgorithm.jl")
+using .BraveNewAlgorithm
+
+# Load required dependencies
+using BlackBoxOptimizationBenchmarking
+include("src/utils.jl")
+include("src/commons.jl")
+
+# 1. Configure the algorithm
+config_parameters = ConfigurationParametersEntity(
+    5,                    # chromosome_size (problem dimensions)
+    30,                   # population_size
+    100,                  # max_generations
+    Dict{String, Int}(    # caste percentages
+        "ALPHA" => 10, "BETA" => 20, "GAMMA" => 30, 
+        "DELTA" => 25, "EPSILON" => 15
+    ),
+    Dict{String, Int}(    # mutation rates per caste
+        "ALPHA" => 5, "BETA" => 8, "GAMMA" => 12, 
+        "DELTA" => 15, "EPSILON" => 20
+    )
+)
+
+# 2. Set up the optimization problem
+fitness_function = FitnessFunction(BlackBoxOptimizationBenchmarking.BBOBFunctions[1], 0)
+search_range = (-5.12, 5.12)
+stopping_criterion = (element, ff) -> element >= ff.fitness_function.f_opt + 1e-8
+
+# 3. Create population model
+population_model = PopulationModel(
+    config_parameters, fitness_function, search_range, stopping_criterion
+)
+
+# 4. Run the algorithm
+generation, results = brave_new_algorithm(population_model)
+
+# 5. Get results
+best_fitness = minimum(results.F_Values)
+println("Best fitness found: $best_fitness")
+println("Completed in $generation generations")
+```
+
+### Using Configuration Files
+
+You can also use JSON configuration files for easier parameter management:
+
+```julia
+# Create config.json
+config = Dict(
+    "CHROMOSOME_SIZE" => 10,
+    "POPULATION_SIZE" => 50,
+    "MAX_GENERATIONS" => 200,
+    "POPULATION_PERCENTAGE" => Dict(
+        "ALPHA" => 15, "BETA" => 25, "GAMMA" => 25, "DELTA" => 20, "EPSILON" => 15
+    ),
+    "MUTATION_RATE" => Dict(
+        "ALPHA" => 3, "BETA" => 6, "GAMMA" => 10, "DELTA" => 15, "EPSILON" => 20
+    )
+)
+
+# Save and load configuration
+using JSON
+open("config.json", "w") do f; JSON.print(f, config, 4); end
+config_parameters = read_parameters_file("config.json")
+
+# Use with algorithm as above...
+```
+
+## 🎯 Supported Optimization Problems
+
+Currently, the algorithm works with **BBOB (Black Box Optimization Benchmarking)** functions:
+
+- **Function 1**: Sphere function (`f(x) = Σx²`)
+- **Function 2**: Rosenbrock function  
+- **Function 3**: Ellipsoid function
+- And more BBOB benchmark functions...
```

Comment:
Please add a link to the BBOB Julia Module we're using here so that the user can consult it.